official/transformer/v2/data_pipeline.py

```diff
@@ -238,10 +244,11 @@ def _read_and_batch_from_files(
 
   if static_batch:
     dataset = dataset.padded_batch(
-        batch_size // max_length, ([max_length], [max_length]),
-        drop_remainder=True)
+        batch_size // num_replicas // max_length * num_replicas,
```

Comment:
is this mainly to avoid an error if they are not exactly divisible?

Reply:
Yes. It's not intuitive to choose a max_length that makes final global bach divisible. Because the final global batch's shape is (global_batch_size // max_length, max_length), and it's not intuitive to choose a max_length that makes (gbs // ml) divisible by num_gpus.

Reply:
Can you use parentheses to make it clear what's being divided by what? it's a bit non obvious from first glance